src/lifecycles/prop.helpers.ts

```diff
@@ -1,10 +1,17 @@
 import * as singleSpa from "../single-spa";
 import { mountParcel } from "../parcels/mount-parcel";
-import { assign } from "../utils/assign";
 import { isParcel, toName } from "../applications/app.helpers";
 import { formatErrorMessage } from "../applications/app-errors";
+import { AppOrParcel, InternalParcel } from "./lifecycle.helpers";
 
-export function getProps(appOrParcel) {
+interface SingleSpaProps {
+  name: string;
+  mountParcel: typeof mountParcel;
+  singleSpa: typeof singleSpa;
+  unmountSelf(): Promise<AppOrParcel>;
+}
+
+export function getProps(appOrParcel: AppOrParcel): SingleSpaProps {
```

Comment:
It might be useful if the return type here was `SingleSpaProps & CustomProps`, although I suppose the real utility would be if that was the type passed to the lifecycle functions itself (since `CustomProps` should be down-castable to the custom props an app expects.

Reply:
👍 updated